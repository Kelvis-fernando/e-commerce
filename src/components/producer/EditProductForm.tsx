import { useEditProductModal } from "../../hooks/producer/useEditProductModal";
import useToast from "../../hooks/useToast";
import {
  AddProductFormContainer,
  InputStyles,
} from "../../styles/producer/addProductForm";
import { ProductCardProps } from "../../types/productCardProps";
import Toast from "../Toast";

interface EditProductModalProps {
  product?: ProductCardProps;
}

export const EditProductModal = ({ product }: EditProductModalProps) => {
  const { handleEditProduct } = useEditProductModal();
  const { showToast, handleCloseToast, handleButtonClick } = useToast();

  return (
    <>
      <AddProductFormContainer
        onSubmit={(event) => {
          handleEditProduct(event, product!.id);
          handleButtonClick();
        }}
      >
        <h1>Cadastre o produto</h1>
        <InputStyles>
          <div>
            <label htmlFor="name">Nome</label>
            <input
              type="text"
              name="name"
              id="name"
              defaultValue={product?.name}
            />
          </div>
          <div>
            <label htmlFor="brand">Marca</label>
            <input
              type="text"
              name="brand"
              id="brand"
              defaultValue={product?.brand}
            />
          </div>
        </InputStyles>
        <InputStyles>
          <div>
            <label htmlFor="type">Tipo</label>
            <select name="type" id="type" defaultValue={product?.type}>
              <option value="fraco">Fraco</option>
              <option value="medio">Medio</option>
              <option value="forte">Forte</option>
            </select>
          </div>
          <div>
            <label htmlFor="price">Preço</label>
            <input
              type="number"
              name="price"
              id="price"
              defaultValue={product?.price}
            />
          </div>
        </InputStyles>
        <InputStyles>
          <div>
            <label htmlFor="classification">Classificação</label>
            <input
              type="text"
              name="classification"
              id="classification"
              defaultValue={product?.classification}
            />
          </div>
          <div>
            <label htmlFor="itensity">Intensidade</label>
            <input
              type="text"
              name="itensity"
              id="itensity"
              defaultValue={product?.itensity}
            />
          </div>
        </InputStyles>
        <InputStyles>
          <div>
            <label htmlFor="origin">Origem</label>
            <input
              type="text"
              name="origin"
              id="origin"
              defaultValue={product?.origin}
            />
          </div>
          <div>
            <label htmlFor="typeToast">Tipo de torra</label>
            <input
              type="text"
              name="typeToast"
              id="typeToast"
              defaultValue={product?.typeToast}
            />
          </div>
        </InputStyles>
        <InputStyles>
          <div>
            <label htmlFor="notes">Notas</label>
            <input
              type="text"
              name="notes"
              id="notes"
              defaultValue={product?.notes}
            />
          </div>
          <div>
            <label htmlFor="quantity">Quantidade em estoque</label>
            <input
              type="number"
              name="quantity"
              id="quantity"
              defaultValue={product?.quantity}
            />
          </div>
        </InputStyles>
        <InputStyles>
          <div>
            <label htmlFor="description">Descrição</label>
            <input
              type="text"
              name="description"
              id="description"
              defaultValue={product?.description}
            />
          </div>
          <div>
            <label htmlFor="image">Image</label>
            <input
              type="text"
              name="image"
              id="image"
              defaultValue={product?.image}
            />
          </div>
        </InputStyles>
        <button type="submit">EDITAR</button>
      </AddProductFormContainer>
      {showToast && (
        <Toast
          typeOfToast="success"
          message="Produto editado com sucesso!"
          onClose={handleCloseToast}
        />
      )}
    </>
  );
};
