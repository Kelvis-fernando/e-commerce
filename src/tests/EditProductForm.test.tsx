import React from "react";
import { render } from "@testing-library/react";
import EditProductForm from "../components/producer/EditProductForm";

describe("EditProductForm", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<EditProductForm />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
