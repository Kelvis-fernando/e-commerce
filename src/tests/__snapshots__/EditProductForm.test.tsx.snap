// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProductForm should render and match snapshot 1`] = `
<div>
  <form
    class="sc-bcXHqe gYwOoV"
  >
    <h1>
      Cadastre o produto
    </h1>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="name"
        >
          Nome
        </label>
        <input
          id="name"
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="brand"
        >
          Marca
        </label>
        <input
          id="brand"
          name="brand"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="type"
        >
          Tipo
        </label>
        <select
          id="type"
          name="type"
        >
          <option
            value="fraco"
          >
            Fraco
          </option>
          <option
            value="medio"
          >
            Medio
          </option>
          <option
            value="forte"
          >
            Forte
          </option>
        </select>
      </div>
      <div>
        <label
          for="price"
        >
          Preço
        </label>
        <input
          id="price"
          name="price"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="classification"
        >
          Classificação
        </label>
        <input
          id="classification"
          name="classification"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="itensity"
        >
          Intensidade
        </label>
        <input
          id="itensity"
          name="itensity"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="origin"
        >
          Origem
        </label>
        <input
          id="origin"
          name="origin"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="typeToast"
        >
          Tipo de torra
        </label>
        <input
          id="typeToast"
          name="typeToast"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="notes"
        >
          Notas
        </label>
        <input
          id="notes"
          name="notes"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="quantity"
        >
          Quantidade em estoque
        </label>
        <input
          id="quantity"
          name="quantity"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      class="sc-gswNZR eKGxfH"
    >
      <div>
        <label
          for="description"
        >
          Descrição
        </label>
        <input
          id="description"
          name="description"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="image"
        >
          Image
        </label>
        <input
          id="image"
          name="image"
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      type="submit"
    >
      EDITAR
    </button>
  </form>
</div>
`;
